src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -45,39 +65,194 @@ type Props = {
+        )
+      );
+    }
+
+    const threadExceptionValues = getThreadException(event, bestThread)?.values;
+
+    return !!(threadExceptionValues
+      ? threadExceptionValues.find(threadExceptionValue =>
+          threadExceptionValue.stacktrace?.frames?.find(frame =>
+            this.isDataMinified(frame.module)
+          )
+        )
+      : bestThread?.stacktrace?.frames?.find(frame => this.isDataMinified(frame.module)));
+  }
+
+  async checkProGuardError() {
     const {event} = this.props;
 
-    if (!event || !event.errors || !(event.errors.length > 0)) {
+    if (!event || event.platform !== 'java') {
+      this.setState({isLoading: false});
       return;
     }
-    const errors = event.errors;
-    const errorTypes = errors.map(errorEntries => errorEntries.type);
-    const errorMessages = errors.map(errorEntries => errorEntries.message);
 
-    this.recordIssueError(errorTypes, errorMessages);
-  }
+    const hasEventErrorsProGuardMissingMapping = event.errors?.find(
+      error => error.type === 'proguard_missing_mapping'
+    );
 
-  shouldComponentUpdate(nextProps: Props) {
-    const {event, showExampleCommit} = this.props;
+    if (hasEventErrorsProGuardMissingMapping) {
+      this.setState({isLoading: false});
+      return;
+    }
 
-    return (
-      (event && nextProps.event && event.id !== nextProps.event.id) ||
-      showExampleCommit !== nextProps.showExampleCommit
+    const proGuardErrors: ProGuardErrors = [];
+
+    const debugImages = event.entries?.find(e => e.type === EntryType.DEBUGMETA)?.data
+      .images as undefined | Array<Image>;
+
+    // When debugImages contains a 'proguard' entry, it must always be only one entry
+    const proGuardImage = debugImages?.find(
+      debugImage => debugImage?.type === 'proguard'
+    );
+
+    const proGuardImageUuid = proGuardImage?.uuid;
+
+    // If an entry is of type 'proguard' and has 'uuid',
+    // it means that the Sentry Gradle plugin has been executed,
+    // otherwise the proguard id wouldn't be in the event.
+    // But maybe it failed to upload the mappings file
+    if (defined(proGuardImageUuid)) {
+      const debugFile = await this.fetchDebugFile(proGuardImageUuid);
+
+      if (!debugFile.length) {
+        proGuardErrors.push({
+          type: 'proguard_missing_mapping',
+          message: projectProcessingIssuesMessages.proguard_missing_mapping,
+          data: {mapping_uuid: proGuardImageUuid},
+        });
+      }
+
+      this.setState({proGuardErrors, isLoading: false});
+      return;
+    } else {
+      if (proGuardImage) {
+        Sentry.withScope(function (s) {
+          s.setLevel(Sentry.Severity.Warning);
+          if (event.sdk) {
+            s.setTag('offending.event.sdk.name', event.sdk.name);
+            s.setTag('offending.event.sdk.version', event.sdk.version);
+          }
+          Sentry.captureMessage('Event contains proguard image but not uuid');
+        });
+      }
+    }
```

Comment:
Oh I just realized we're not showing an error in this case, where no `debug_meta` exists at all.
This would also indicate the plugin wasn't installed at all.
Could we also have a `captureMessage` here so we can see how this performs before we add the error? (This is where false positive could trigger: No Proguard is used at all, but a frame did match the regex)

Reply:
Bruno I have added this condition 

https://github.com/getsentry/sentry/blob/1a6875ae62a4270e64dc97dd7286ea5fcc5d83d3/src/sentry/static/sentry/app/components/events/eventEntries.tsx#L247

Please is that what you meant?

I also added the  `s.setTag`, so then we can know more details about the `sdk` in case this capture is fired.

Reply:
Sounds good.